corehq/apps/saved_reports/tests/test_models.py

```diff
@@ -0,0 +1,34 @@
+from django.test import SimpleTestCase
+from unittest.mock import create_autospec
+from corehq.apps.users.models import WebUser
+from ..models import ReportNotification
+
+
+class TestReportNotification(SimpleTestCase):
+    def test_unauthorized_user_cannot_view_report(self):
+        report = ReportNotification(owner_id='5', domain='test_domain', recipient_emails=[])
+        bad_user = self._create_user(id='3', is_domain_admin=False)
+        self.assertFalse(report.can_be_viewed_by(bad_user))
+
+    def test_owner_can_view_report(self):
+        report = ReportNotification(owner_id='5', domain='test_domain', recipient_emails=[])
+        owner = self._create_user(id='5')
+        self.assertTrue(report.can_be_viewed_by(owner))
+
+    def test_domain_admin_can_view_report(self):
+        report = ReportNotification(owner_id='5', domain='test_domain', recipient_emails=[])
+        domain_admin = self._create_user(is_domain_admin=True)
+        self.assertTrue(report.can_be_viewed_by(domain_admin))
+
+    def test_subscribed_user_can_view_report(self):
+        report = ReportNotification(owner_id='5', domain='test_domain', recipient_emails=['test@dimagi.com'])
+        subscribed_user = self._create_user(email='test@dimagi.com')
+        self.assertTrue(report.can_be_viewed_by(subscribed_user))
+
+    def _create_user(self, id='100', email='not-included', is_domain_admin=False):
+        user_template = WebUser(domain='test_domain', username='test_user')
+        user = create_autospec(user_template, spec_set=True, _id=id)
```

Comment:
why does this need to be a mock? I don't see anywhere that's making use of any mock features.

Reply:
Is the question why these tests aren't using a real `WebUser`, or why I'm preferring a mock here vs rolling my own fake user? For the former, in order to handle the situation where the user is a domain admin, I'd need to, at a minimum, create and delete a couch domain, which is overkill for the intention of this test (I'm just trying to verify the boolean logic). As written, I can use `SimpleTestCase` instead of `TestCase`, which lets the test run in milliseconds.

If the question was why am I using a mock rather than rolling my own fake, it's because the mock gives some features to ensure I'm not just creating a mock which will always pass the tests. Speccing the mock means that if the methods this test depends on were to change in the future, this test should fail.

Reply:
I checked this out locally and I see the real reason you have to mock is because JsonObject won't allow you to assign a lambda to attributes.